maven-embedder/src/main/java/org/apache/maven/cli/CLIManager.java

```diff
@@ -123,7 +123,7 @@ public CLIManager()
         options = new Options();
         options.addOption( Option.builder( Character.toString( HELP ) ).longOpt( "help" ).desc( "Display help information" ).build() );
         options.addOption( Option.builder( Character.toString( ALTERNATE_POM_FILE ) ).longOpt( "file" ).hasArg().desc( "Force the use of an alternate POM file (or directory with pom.xml)" ).build() );
-        options.addOption( Option.builder( Character.toString( SET_SYSTEM_PROPERTY ) ).hasArg().desc( "Define a system property" ).build() );
+        options.addOption( Option.builder( Character.toString( SET_SYSTEM_PROPERTY ) ).numberOfArgs( 2 ).valueSeparator( '=' ).desc( "Define a system property" ).build() );
```

Comment:
Interesting, does the documentation really say 2?

Reply:
No. The Commons CLI documentation says to do something different, but doing that would "eat" the remaining arguments; as a result, "mvn -Dx=y verify" would fail with "No goals have been specified for this build".

Reply:
Instane, this is worth opening an issue with Commons CLI.

Reply:
Thought about it... Given how much attention went into [CLI-312](https://issues.apache.org/jira/browse/CLI-312) from Commons side, I doubt it will be of use to file an issue. The only people that interacted on that ticket were you, Robert and me.

Reply:
Agree. CLI is on the downfall. It is good enough for Maven 4, I guess.